main.cpp

```diff
@@ -18,13 +18,22 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private:
+    Rewriter& rewriter;
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : rewriter(rewriter) {};
 
     void run(const MatchFinder::MatchResult &Result) override {

```

Comment:
- How to prevent parentheses from being added if they are already present in the (int)(f) code?
- How to prevent changing the C style cast code for a special case - when we want to suppress warnings about unused variables?
- How to prevent changing the C style cast code in macros?

Please provide code examples.

Reply:
1. After replacing the text on the right border (34 line) add a condition check that the expression is wrapped in brackets. This is possible by running the following code:
`if(!isa<ParenExpr>(Expr->getSubExprAsWritten())) {
  rewriter.InsertText(Expr->getSubExprAsWritten()->getBeginLoc(), "(");
  rewriter.InsertTextAfterToken(endLoc, ")");
}`
This code checks the condition that subexpression belongs to the class ParenExpr. If this condition is not in class ParenExpr than add brackets at the beginning and end of the expression.
2. Change 28 line like this:
`if (Expr) && (Expr->getCastKind() != CK_ToVoid)`
This is a condition for the conversion type. If it is void, then we do nothing.
3. Change 28 line like this:
`if (Expr) && (!(cStyleCExpr->getExprLoc().isMacroID()))`
By analogy with point 2: this is a condition for checking for a macros.